sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -528,6 +528,28 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
 }
 
 
+/**
+ * Push the foldable expression into (if / case) branches.
+ */
+object PushFoldableIntoBranches extends Rule[LogicalPlan] with PredicateHelper {
+  def apply(plan: LogicalPlan): LogicalPlan = plan transform {
```

Comment:
nit: we cannot use `transformAllExpressions` here?

Reply:
This copied from https://github.com/apache/spark/blob/8ccc3c1f8804e34b46fd9bec19a017db61b37f54/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala#L474

Reply:
Ah, I see. I think its okay to simply use `transformAllExpressions` here.